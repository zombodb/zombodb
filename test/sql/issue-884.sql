CREATE TABLE issue884
(
    id       SERIAL8 NOT NULL PRIMARY KEY,
    name     text    NOT NULL,
    testdate date
);
CREATE INDEX es_idxhighlighteddatetest
    ON issue884 USING zombodb ((issue884.*));

INSERT INTO issue884(name, testdate)
SELECT 'testrow ' || x, '2024-01-05'::date - (x || ' days')::interval from generate_series(1, 100) x;

-- date field
SELECT issue884.*
FROM issue884
WHERE issue884 ==> 'testdate = "2023-12-10" /TO/ "2023-12-11"';

SELECT issue884.*, id AS primary_key
FROM issue884
         INNER JOIN LATERAL
    zdb.highlight_document('issue884'::regclass,
                           to_json(issue884),
                           'testdate = "2023-12-10" /TO/ "2023-12-11"'::TEXT) ON TRUE
ORDER BY "primary_key", "field_name", "position";


-- id field
SELECT issue884.*
FROM issue884
WHERE issue884 ==> 'id = 27 /TO/ 31';

SELECT issue884.*, id AS primary_key
FROM issue884
         INNER JOIN LATERAL
    zdb.highlight_document('issue884'::regclass,
                           to_json(issue884),
                           'id = 27 /TO/ 31'::TEXT) ON TRUE
ORDER BY "primary_key", "field_name", "position";

DROP TABLE issue884;