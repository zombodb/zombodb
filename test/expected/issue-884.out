CREATE TABLE issue884
(
    id       SERIAL8 NOT NULL PRIMARY KEY,
    name     text    NOT NULL,
    testdate date
);
CREATE INDEX es_idxhighlighteddatetest
    ON issue884 USING zombodb ((issue884.*));
INSERT INTO issue884(name, testdate)
SELECT 'testrow ' || x, '2024-01-05'::date - (x || ' days')::interval from generate_series(1, 100) x;
-- date field
SELECT issue884.*
FROM issue884
WHERE issue884 ==> 'testdate = "2023-12-10" /TO/ "2023-12-11"';
 id |    name    |  testdate  
----+------------+------------
 25 | testrow 25 | 12-11-2023
 26 | testrow 26 | 12-10-2023
(2 rows)

SELECT issue884.*, id AS primary_key
FROM issue884
         INNER JOIN LATERAL
    zdb.highlight_document('issue884'::regclass,
                           to_json(issue884),
                           'testdate = "2023-12-10" /TO/ "2023-12-11"'::TEXT) ON TRUE
ORDER BY "primary_key", "field_name", "position";
 id |    name    |  testdate  | primary_key 
----+------------+------------+-------------
 25 | testrow 25 | 12-11-2023 |          25
 26 | testrow 26 | 12-10-2023 |          26
(2 rows)

-- id field
SELECT issue884.*
FROM issue884
WHERE issue884 ==> 'id = 27 /TO/ 31';
 id |    name    |  testdate  
----+------------+------------
 27 | testrow 27 | 12-09-2023
 28 | testrow 28 | 12-08-2023
 29 | testrow 29 | 12-07-2023
 30 | testrow 30 | 12-06-2023
 31 | testrow 31 | 12-05-2023
(5 rows)

SELECT issue884.*, id AS primary_key
FROM issue884
         INNER JOIN LATERAL
    zdb.highlight_document('issue884'::regclass,
                           to_json(issue884),
                           'id = 27 /TO/ 31'::TEXT) ON TRUE
ORDER BY "primary_key", "field_name", "position";
 id |    name    |  testdate  | primary_key 
----+------------+------------+-------------
 27 | testrow 27 | 12-09-2023 |          27
 28 | testrow 28 | 12-08-2023 |          28
 29 | testrow 29 | 12-07-2023 |          29
 30 | testrow 30 | 12-06-2023 |          30
 31 | testrow 31 | 12-05-2023 |          31
(5 rows)

DROP TABLE issue884;
